src/slider.ts

```diff
@@ -0,0 +1,1331 @@
+    this.#startDragging(event, this.#minimumHandleElementRef.value!);
+  }
+
+  #onMinimumInputChange() {
+    if (this.#minimumInputElementRef.value && this.maximumValue !== undefined) {
+      const inputValue = Number(this.#minimumInputElementRef.value.value);
+      const normalizedValue = Math.round(inputValue / this.step) * this.step;
+
+      this.minimumValue = Math.min(
+        Math.max(normalizedValue, this.min),
+        // Ensures the minimum value always stays at least one step below the
+        // maximum value, maintaining the required separation.
+        // Without this, users could set the minimum value equal to or higher
+        // than the maximum value, creating issues, both visually and
+        // operationally.
+        //
+        // Consider the case where maximumValue is 75 and step is 5.
+        // Without this, a user could set the value to 80, which would
+        // be invalid. With this constraint, the minimum value can't go higher
+        // than 70 (75-5).
+        this.maximumValue - this.step,
+      );
+
+      this.#updateHandlesAndTrack();
+
+      // Unlike "input" events, "change" events aren't composed. So we have to
+      // manually dispatch them.
+      this.dispatchEvent(
+        new Event('change', { bubbles: true, composed: true }),
+      );
+    }
+  }
+
+  #onMinimumInputInput() {
+    if (this.#minimumInputElementRef.value) {
+      const inputValue = Number(this.#minimumInputElementRef.value.value);
+
+      if (this.maximumValue !== undefined && inputValue <= this.maximumValue) {
+        this.minimumValue = Math.max(inputValue, this.min);
+        this.#updateHandlesAndTrack();
+      }
+    }
+  }
+
+  #onMultipleHandleKeydown(event: KeyboardEvent, handle: HTMLElement) {
+    if (this.disabled || this.readonly) {
+      return;
+    }
+
+    if (this.minimumValue !== undefined && this.maximumValue !== undefined) {
+      const isMinimumHandle = handle === this.#minimumHandleElementRef.value;
+
+      let value = isMinimumHandle ? this.minimumValue : this.maximumValue;
+
+      switch (event.key) {
+        case 'ArrowLeft':
+        case 'ArrowDown': {
+          value = value - this.step;
+          break;
+        }
+        case 'ArrowRight':
+        case 'ArrowUp': {
+          value = value + this.step;
+          break;
+        }
+        case 'PageDown': {
+          value = value - this.step * 10;
+          break;
+        }
+        case 'PageUp': {
+          value = value + this.step * 10;
+          break;
+        }
+        case 'Home': {
+          value = isMinimumHandle ? this.min : this.minimumValue + this.step;
+          break;
+        }
+        case 'End': {
+          value = isMinimumHandle ? this.maximumValue - this.step : this.max;
+          break;
+        }
+        case 'Enter': {
+          this.form?.requestSubmit();
+          return;
+        }
+
+        default: {
+          return;
+        }
+      }
+
+      event.preventDefault();
```

Comment:
Is it obvious why the event is canceled?

Reply:
Comment added (thank you!).